k8s-scheduler/src/main/java/com/vmware/dcm/ScopedModel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-2
+ */
+
+package com.vmware.dcm;
+
+import com.vmware.dcm.k8s.generated.Tables;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SortField;
+import org.jooq.Table;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.selectFrom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * A Scope to reduce the Model's solver input.
+ */
+
+public class ScopedModel {
+    private static final Logger LOG = LoggerFactory.getLogger(ScopedModel.class);
+    private final DSLContext conn;
+    private final Model model;
+
+    // dynamically adjuct number of scoped nodes with limitTune
+    private double limitTune = 1.0;
```

Comment:
For now, these three parameters should be `private static final` and named XYZ_DEFAULT.